RLTest/loader.py

```diff
@@ -96,48 +101,53 @@ def load_spec(self, arg):
             sys.path.append(dirname)
 
         module_name, _ = os.path.splitext(os.path.basename(filename))
+        toplevel_filter = None
+        subfilter = None
         if varname:
```

Comment:
Exception throwing is cheap (preferred even) in python.

```
if varname is not None:
    try:
        toplevel_filter, subfilter = varname.split('.')
    except ValueError:
        toplevel_filter=  varname
```

Reply:
except that split() doesn't throw.

Reply:
Pasted the wrong bit:
```

Reply:
Pasted the wrong piece - sorry:
```
try:
    toplevel_filter = varname.split('.')[0]
    subfilter = varname.split('.')[1]
except IndexError:
    toplevel_filter=  varname
except AttributeError:  # None case
    pass
```

^^ or similar.

Reply:
Note that `"abc".split('.') == ['abc']` - no exception here.

But anyways, which one looks clearer?
```
    toplevel_filter, subfilter  = None, None
    if varname:
        if '.' in varname:
            toplevel_filter, subfilter = varname.split('.')
        else:
            toplevel_filter = varname
```
or
```
try:
    x = varname.split('.')
    toplevel_filter = x[0]
    subfilter = x[1]
except IndexError:
    toplevel_filter = varname
except AttributeError:  # None case
    pass
```